lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -109,7 +104,7 @@ namespace Js
 
         // Mark module as root module if it currently has no parents
         // Note, if there are circular imports it may gain parents later
-        if (parentModuleList == nullptr)
+        if (parentModuleList == nullptr && promise == nullptr)
```

Comment:
I had to restore the some of the separation between dynamic imports and other rootModules to avoid an issue with dynamically imported modules that error being parsed twice.